packages/profile-sync-controller/src/shared/env.ts

```diff
@@ -39,14 +39,29 @@ const ENV_URLS: Record<Env, EnvUrlsEntry> = {
  * Validates and returns correct environment endpoints
  *
  * @param env - environment field
- * @returns the correct environment url
+ * @param options - optional configuration
+ * @param options.authApiVersion - optional version suffix for the auth API URL (e.g., 'v2')
+ * @returns the correct environment urls
  * @throws on invalid environment passed
  */
-export function getEnvUrls(env: Env): EnvUrlsEntry {
+export function getEnvUrls(
+  env: Env,
+  options?: { authApiVersion?: string },
+): EnvUrlsEntry {
   if (!ENV_URLS[env]) {
     throw new Error('invalid environment configuration');
   }
-  return ENV_URLS[env];
+
+  const urls = ENV_URLS[env];
+
+  if (options?.authApiVersion) {
+    return {
+      ...urls,
+      authApiUrl: `${urls.authApiUrl}/${options.authApiVersion}`,
```

Comment:
Just double-checking, we're not embedding any sort of GET parameters in `authApiUrl` here? Otherwise the URL could be wrongly formatted in this case?

Reply:
Looks like `ENV_URLS` is actually just a constant mapping, mb... I thought we were somehow injecting those URLs from some env vars but nope 😁

And after discussing with @mathieuartu, we already are doing this kind of string-concatenation in other places: https://github.com/MetaMask/core/blob/ee3a8c0fd8ebb1622c512221e47569e9540327e9/packages/profile-sync-controller/src/sdk/authentication-jwt-bearer/services.ts#L80-L96